src/main/java/org/isf/pregnancy/gui/PregnancyBrowser.java

```diff
@@ -0,0 +1,856 @@
+	private int TOTAL_ROWS;
+	private Patient patient;
+	private JButton jSearchButton;
+	private JTextField searchPatientTextField;
+	private List<JLabel> deltypeLabel;
+	private List<JLabel> deltypeResLabel;
+	private String lastKey;
+	private DefaultTableModel model;
+
+	/**
+	 * Constructor called from the main menu
+	 */
+	public PregnancyBrowser() {
+		setTitle(MessageBundle.getMessage("angal.pregnancy.patientsbrowser.title"));
+		myFrame = this;
+		initComponents();
+		pack();
+		setLocationRelativeTo(null);
+		this.setVisible(true);
+		myFrame.addWindowListener(new WindowAdapter() {
+			public void windowClosing(WindowEvent e) {
+				// to free memory
+				if (pregnancyPatientList != null)
+					pregnancyPatientList.clear();
+				dispose();
+			}
+		});
+	}
+
+	/**
+	 * constructor for the AdmissionBrowser to see only the pregnancyvisits for the
+	 * patient
+	 * 
+	 * @param admittedpatient the admitted patient
+	 */
+	public PregnancyBrowser(Patient admittedpatient) {
+		setTitle(MessageBundle.getMessage("angal.pregnancy.patientsbrowser.title"));
+		myFrame = this;
+		pregnancyPatientList = new ArrayList<AdmittedPatient>();
+		this.patient = admittedpatient;
+		initComponents();
+		pack();
+		setLocationRelativeTo(null);
+		this.setVisible(true);
+		myFrame.addWindowListener(new WindowAdapter() {
+			public void windowClosing(WindowEvent e) {
+				// to free memory
+				if (pregnancyPatientList != null)
+					pregnancyPatientList.clear();
+				dispose();
+			}
+		});
+
+	}
+
+	/**
+	 * intis the components
+	 */
+	private void initComponents() {
+		getContentPane().add(getPatientPanel(), BorderLayout.NORTH);
+		getContentPane().add(getVisitPanel(), BorderLayout.CENTER);
+		getContentPane().add(getPregnancyButtonPanel(), BorderLayout.SOUTH);
+	}
+
+	private JPanel getPatientPanel() {
+		JPanel dataPatientListPanel = new JPanel(new BorderLayout());
+		JPanel navigation = new JPanel(new FlowLayout(FlowLayout.CENTER));
+		under = new JLabel(MessageBundle.getMessage("angal.common.page.label"));
+		under.setPreferredSize(new Dimension(60, 21));
+		navigation.add(getPreviousButton()).setPreferredSize(new Dimension(100, 21));
+		navigation.add(getComboBoxPages());
+		navigation.add(under);
+		navigation.add(getNextButton()).setPreferredSize(new Dimension(100, 21));
+		dataPatientListPanel.add(navigation, BorderLayout.NORTH);
+		dataPatientListPanel.add(getSearchPanel(), BorderLayout.WEST);
+		dataPatientListPanel.add(getPatientScrollPane(), BorderLayout.CENTER);
+		dataPatientListPanel.add(getPatientButtonPanel(), BorderLayout.EAST);
+		return dataPatientListPanel;
+	}
+
+	public JButton getNextButton() {
+		if (next == null) {
+			next = new JButton(MessageBundle.getMessage("angal.visit.nextarrow.btn"));
+			next.setPreferredSize(new Dimension(30, 21));
+			next.setMnemonic(MessageBundle.getMnemonic("angal.visit.nextarrow.btn.key"));
+			next.addActionListener(actionEvent -> {
+				if (!previous.isEnabled())
+					previous.setEnabled(true);
```

Comment:
```suggestion
				if (!previous.isEnabled()) {
					previous.setEnabled(true);
				}
```
Please find all the places where you don't put accolades and put it